### PR TITLE
Update to OpenJDK-11 RMB-328

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java [hunk 1]
```diff
@@ -92,8 +92,12 @@
...
    */
   public CQL2PgJSON(String field) throws FieldException {
-    doInit(field, null);
-    initDbTable();
+    try {
+      doInit(field, null);
+      initDbTable();
+    } catch (NullPointerException e) {
+      logger.log(Level.SEVERE, e.getMessage(), e);
+    }
```
Comment (julianladisch) [inline cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java:100]: Please add a comment that explains why this exception is logged without failing the constructor method.
Can you add a unit test for it?

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+import org.apache.commons.collections4.MultiMapUtils;
+import org.apache.commons.collections4.SetValuedMap;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.SystemUtils;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+
+public class ClassPath {
```
Comment (julianladisch) [inline domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:37]: Can you add the source where this has been copied from, the copyright attribution, and the license?

File: doc/upgrading.md [hunk 3]
```diff
@@ -15,6 +15,15 @@ See the [NEWS](../NEWS.md) summary of changes for each version.
 * [Version 25](#version-25)
 * [Version 20](#version-20)
 
+## Version 31
+
+* [RMB-328](https://issues.folio.org/browse/RMB-328) Update to OpenJDK 11.
+  A few updates to `pom.xml` are necessary. For `maven-compiler-plugin`
+  update `version` to `3.8.1`. and use `<release>11</release>` instead
+  of `source` and `target` elements.
+  Update aspectj version to 1.9.5. Update `aspect-maven-plugin` with
```
Comment (julianladisch) [inline doc/upgrading.md:24]: `aspect-maven-plugin` should be `aspectj-maven-plugin`

File: doc/upgrading.md [hunk 4]
```diff
@@ -15,6 +15,15 @@ See the [NEWS](../NEWS.md) summary of changes for each version.
 * [Version 25](#version-25)
 * [Version 20](#version-20)
 
+## Version 31
+
+* [RMB-328](https://issues.folio.org/browse/RMB-328) Update to OpenJDK 11.
+  A few updates to `pom.xml` are necessary. For `maven-compiler-plugin`
+  update `version` to `3.8.1`. and use `<release>11</release>` instead
```
Comment (julianladisch) [inline doc/upgrading.md:22]: remove . after `3.8.1`

File: doc/upgrading.md [hunk 5]
```diff
@@ -15,6 +15,15 @@ See the [NEWS](../NEWS.md) summary of changes for each version.
...
+  of `source` and `target` elements.
+  Update aspectj version to 1.9.5. Update `aspect-maven-plugin` with
+  groupId `com.nickwongdev` and version `1.12.6`.
```
Comment (julianladisch) [inline doc/upgrading.md:25]: Can you add a link to an [example](https://github.com/folio-org/mod-configuration/pull/87/files)?

Comment (julianladisch) [inline doc/upgrading.md:25]: Can you also mention to add `buildNode = 'jenkins-agent-java11'` to Jenkinsfile, change `folioci/alpine-jre-openjdk8:latest` to `folioci/alpine-jre-openjdk11:latest` in Dockerfile and to remove docker/docker-entrypoint.sh?

File: cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java (hunk 1 above)
Comment (adamdickmeiss) [reply cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java:100]: I'll just remove this again. An exception was thrown when classes was not found and it was not logged in whatever parent was calling this. That's why it was added.

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java (hunk 2 above)
Comment (adamdickmeiss) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:37]: That would be something @dmtkachenko presumably knows.

File: doc/upgrading.md (hunk 3 above)
Comment (adamdickmeiss) [reply doc/upgrading.md:24]: yep

File: doc/upgrading.md (hunk 4 above)
Comment (adamdickmeiss) [reply doc/upgrading.md:22]: yep

File: doc/upgrading.md (hunk 5 above)
Comment (adamdickmeiss) [reply doc/upgrading.md:25]: Not examples yet, since no modules are yet merged.

Comment (adamdickmeiss) [reply doc/upgrading.md:25]: What the problem with entrypoint.sh?

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java (hunk 2 above)
Comment (adamdickmeiss) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:53]: I copied the licence from file that this is based upon (AFAIK)

Comment (dmtkachenko) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:53]: Yes, correct. It was originally copied and modified from
https://github.com/google/guava/blob/v29.0/guava/src/com/google/common/reflect/ClassPath.java

File: doc/upgrading.md (hunk 5 above)
Comment (julianladisch) [reply doc/upgrading.md]: Doesn't the linked pull request qualify as an example?

Comment (julianladisch) [reply doc/upgrading.md]: Neither alpine-jre-openjdk8 nor alpine-jre-openjdk11 use it, and upgrading.md is missing a suggestion to remove it. This is a good opportunity to clean up.

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java (hunk 2 above)
Comment (julianladisch) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:53]: This copy still violates the requirements of the Apache License.

File: doc/upgrading.md (hunk 5 above)
Comment (adamdickmeiss) [reply doc/upgrading.md]: You want the upgrade manual to point to this huge PR itself ?

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java (hunk 2 above)
Comment (TarasSpashchenko) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:53]: How does it violate the License requirements?

File: doc/upgrading.md (hunk 5 above)
Comment (adamdickmeiss) [reply doc/upgrading.md]: Or do you mean this (shorter) one: https://github.com/folio-org/mod-inventory-storage/pull/485

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java (hunk 2 above)
Comment (adamdickmeiss) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:53]: In what way? ClassPath is Apache 2.0. So is RMB

Comment (julianladisch) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:56]: If changed the notice to comply with this license requirement: "You must cause any modified files to carry prominent notices stating that You changed the files"

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java [hunk 6]
```diff
@@ -0,0 +1,566 @@
...
+          } catch (SecurityException e) { // File.toURI checks to see if the file is a directory
+            urls.add(new URL("file", null, new File(entry).getAbsolutePath()));
+          }
+        } catch (MalformedURLException e) {
+          log.warn("Malformed classpath entry: " + entry, e);
+        }
+      }
+
+      return List.copyOf(urls);
+    }
```
Comment (julianladisch) [inline domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:464]: ClassPathTest.testGetClassPathUrls fails.

File: doc/upgrading.md (hunk 5 above)
Comment (julianladisch) [reply doc/upgrading.md]: Yes, https://github.com/folio-org/mod-inventory-storage/pull/485/files is slightly more simple than https://github.com/folio-org/mod-configuration/pull/87/files I proposed above.

File: domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java (hunk 6 above)
Comment (adamdickmeiss) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:464]: I have two tests failing.. When I run them individually they all succeed. SO not isolated. I'm inclined to drop the test unless somebody has an easy fix to provoke *some* order that works.

Comment (julianladisch) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:462]: I would rather fix the implementation bugs.

Comment (julianladisch) [reply domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java:462]: Implementation bugs fixed.

**Pull request merged**